### PR TITLE
Adds dev flag for new private key

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
Date: 27 Jan 2024

Developer Name: @Ajeyakrishna-k 

----

Issue: https://github.com/Real-Dev-Squad/cron-jobs/issues/53

## Description

Adds dev flag for new private key

**Documentation Updated?**

-   [ ] Yes
-   [x] No

**Breaking Changes**

-   [ ] Yes
-   [x] No

**Development Tested?**

-   [x] Yes
-   [ ] No

**Tested in Staging?**

-   [ ] Yes
-   [x] No

**Under Feature Flag**

-   [x] Yes
-   [ ] No

**Database Changes**

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Review by skv93-coder — APPROVED** [review]

**Review by ajoykumardas12 — APPROVED** [review]

File: src/services/discordBotServices.ts [hunk 1]
```diff
@@ -6,8 +6,9 @@ export const updateUserRoles = async (env: env, payload: DiscordUserRole[]): Pro
 	try {
 		const url = config(env).DISCORD_BOT_API_URL;
 		const token = await generateDiscordBotJwt(env);
-
-		const response = await env.DISCORD_BOT.fetch(`${url}/roles?action=add-role`, {
+		//TODO(@Ajeyakrishna-k): remove dev flag https://github.com/Real-Dev-Squad/discord-slash-commands/issues/193
+		const devQuery = env.DEV ? '&dev=true' : '';
```
Comment (ankushdharkar) [inline src/services/discordBotServices.ts]: What is `env.DEV`?

Comment (Ajeyakrishna-k) [reply src/services/discordBotServices.ts]: its used as feature flag.

Comment (ankushdharkar) [reply src/services/discordBotServices.ts]: Could we make it more descriptive please? Maybe something like "FEATURE_SETUP_OWN_PRIVATE_KEY_... " etc

Comment (Ajeyakrishna-k) [reply src/services/discordBotServices.ts]: changed it to `FF_SELF_PRIVATE_KEY`

**Review by Achintya-Chatterjee — APPROVED** [review]

File: src/services/discordBotServices.ts (hunk 1 above)
Comment (prakashchoudhary07) [reply src/services/discordBotServices.ts]: changed it to `FF_CRON_DISCORD_KEY_PAIR_FLOW`

**Pull request merged**